src/Utilities/Compiler/Options/CategorizedAnalyzerConfigOptions.cs

```diff
@@ -113,7 +113,7 @@ public T GetOptionValue<T>(string optionName, DiagnosticDescriptor rule, TryPars
                 return defaultValue;
             }
 
-            return (T)_computedOptionValuesMap.GetOrAdd($"{rule.Id}.{optionName}", _ => ComputeOptionValue(optionName, rule, tryParseValue, defaultValue))!;
+            return (T)_computedOptionValuesMap.GetOrAdd($"{rule.Id}.{optionName}.{defaultValue?.GetHashCode()}", _ => ComputeOptionValue(optionName, rule, tryParseValue, defaultValue))!;
```

Comment:
I had to update the cache to allow callers with a different "forced value" to have a different result and not the same cache entry. We might want to have a constraint on `T` if we go-on with the `GetHashCode` method. Alternatively, we could change the method signature to have the default value being a `string` and not a `T` and we would call `TryParse` on this `string` entry.

Anyways let me know which direction you think is best.

Reply:
Yeah, this looks fragile. I'd rather avoid all changes in this PR to any of the options infrastructure and just make an isolated fix to Dispose analysis, as suggested in https://github.com/dotnet/roslyn-analyzers/pull/3299/files#r384039514